control/statesp.py

```diff
@@ -885,8 +889,12 @@ def sample(self, Ts, method='zoh', alpha=None):
             raise ValueError("System must be continuous time system")
 
         sys = (self.A, self.B, self.C, self.D)
-        Ad, Bd, C, D, dt = cont2discrete(sys, Ts, method, alpha)
-        return StateSpace(Ad, Bd, C, D, dt)
+        if prewarp_frequency is not None:
+            Twarp = 2*np.tan(prewarp_frequency*Ts/2)/prewarp_frequency
+            Ad, Bd, C, D, _ = cont2discrete(sys, Twarp, method, alpha)
+        else:
+            Ad, Bd, C, D, _ = cont2discrete(sys, Ts, method, alpha)
+        return StateSpace(Ad, Bd, C, D, Ts)
```

Comment:
This could be shortened to
```
Twarp = Ts if prewarp_frequency is None else 2*np.tan(prewarp_frequency*Ts/2)/prewarp_frequency
Ad, Bd, C, D, _ = cont2discrete(sys, Twarp, method, alpha)
```

Reply:
This might work except that we need to keep the original Ts because it gets used a few lines later to construct the discrete time system that the function returns.

Reply:
Never mind, I think you're right.

Reply:
fixed a failing test and realized I should add an additional condition that prewarping will only apply to bilinear transformation.